src/librados/IoCtxImpl.cc

```diff
@@ -1389,6 +1423,45 @@ int librados::IoCtxImpl::read(const object_t& oid,
   return bl.length();
 }
 
+int librados::IoCtxImpl::repair_copy(const object_t& oid,
+				     uint64_t ver, uint32_t what,
+				     const std::vector<pg_shard_t>& bad_shards,
+				     epoch_t epoch)
+{
+  ::ObjectOperation op;
+  op.assert_interval(epoch);
+  op.assert_version(ver);
+  op.repair_copy(what, bad_shards);
+  return operate(oid, &op, nullptr, CEPH_OSD_FLAG_REPAIR_WRITES);
+}
+
+int librados::IoCtxImpl::aio_repair_copy(const object_t& oid, AioCompletionImpl *comp,
+					 uint64_t ver, uint32_t what,
+					 const std::vector<pg_shard_t>& bad_shards,
+					 epoch_t epoch)
+{
+  ::ObjectOperation op;
+  op.assert_interval(epoch);
+  op.assert_version(ver);
+  op.repair_copy(what, bad_shards);
+
+  auto onack = new C_aio_Complete(comp);
+
+  comp->io = this;
+  queue_aio_write(comp);
+
+  auto objecter_op = objecter->prepare_mutate_op(oid, oloc, op, snapc,
```

Comment:
`Objecter::init_ops()` prepend the ops in given `ObjectOperation` into the new ops.  like
```
[extra,extra,extra,to-be-filled]
```
`prepare_read_op()` fills up the last "to-be-filled slot with a newly created read op.

but in the repair-read case, what we want is
```
[assert-interval,extra,extra,extra]
```
where extra is the read ops passed in by caller. to use `init_ops()` we need to create an `ObjectOperation`, and put assert-interval into it first, and copy the ops in `ObjectOperation` to the ops filled by `init_ops()`. i don't see the advantage of this approach.